nodejs/eks/cluster.ts

```diff
@@ -340,6 +340,20 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         throw new Error("Setting nodeGroupOptions, and any set of singular node group option(s) on the cluster, is mutually exclusive. Choose a single approach.");
     }
 
+    // Check if the cluster name provided or autoname used starts with 'eks-'.
+    // The cluster name is used for the fargate profile name, and we must
+    // ensure `eks-` is not used in either naming option, as AWS considers it
+    // a reserved prefix and will fail the fargate profile creation.
+    if (args.fargate) {
+        const nameToCheck = args.name || name.concat("-");
```

Comment:
Do you  need `pulumi.concat` here? I don't think the basic concat works on some Inputs

Reply:
The `name` that is being checked and concatenated is from the Cluster ctor, which is of type `string`. What inputs do you think would not work?

https://github.com/pulumi/pulumi-eks/blob/7dd282665e72bf8af3f292c1354bedc4be2542e7/nodejs/eks/cluster.ts#L1281-L1301

Reply:
Oops, you're right. I misread it as `args.name`